tests/modeladmin/models.py

```diff
@@ -28,6 +28,39 @@ class Concert(models.Model):
     ), blank=True)
 
 
+@python_2_unicode_compatible
```

Comment:
I think you could combine the tests and [inline the models](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/unit-tests/#isolating-model-registration) (since no database queries are required here). The `__str__` methods and `@python_2_unicode_compatible` also don't seem needed now.